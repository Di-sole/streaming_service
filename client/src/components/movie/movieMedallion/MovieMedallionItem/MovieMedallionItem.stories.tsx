import { Meta, StoryObj } from '@storybook/react';
import MovieMedallionItem from './MovieMedallionItem';
import './MovieMedallionItem.module.scss';

const meta: Meta<typeof MovieMedallionItem> = {
    title: 'Cards/Medallion',
    component: MovieMedallionItem,
    argTypes: {
        disabled: {
            control: false,
        },
    },
};

export default meta;

type Story = StoryObj<typeof MovieMedallionItem>;

export const Active: Story = {
    decorators: [
        (Story) => (
            <div style={{ background: '#100e19', padding: '10px 10px 0px', cursor: 'pointer', width: '100px' }}>
                <Story />
            </div>
        ),
    ],
    args: {
        text: 'Текст',
        children: <div></div>,
    },
};

export const inActive: Story = {
    decorators: [
        (Story) => (
            <div style={{ background: '#100e19', padding: '10px 10px 0px', width: '100px' }}>
                <Story />
            </div>
        ),
    ],
    args: {
        text: 'Текст',
        children: <div></div>,
        disabled: true,
    },
};
