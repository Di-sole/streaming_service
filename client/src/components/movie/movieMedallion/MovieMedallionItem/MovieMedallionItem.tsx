import styles from './MovieMedallionItem.module.scss';

interface MovieMedallionItemProps {
    children: React.ReactNode;
    text: string;
    disabled?: boolean;
}

const MovieMedallionItem = ({ children, text, disabled = false }: MovieMedallionItemProps) => {
    return (
        <div
            className={[styles.container, disabled ? styles.container_inactive : ''].join(' ').trim()}
            data-testid="medallionItem"
        >
            <div className={styles.container__content}>
                <div className={styles.container__contentBlock}>{children}</div>
            </div>
            <div className={styles.container__caption}>{text}</div>
        </div>
    );
};

export default MovieMedallionItem;
