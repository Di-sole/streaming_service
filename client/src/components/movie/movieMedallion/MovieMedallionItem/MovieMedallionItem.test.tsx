import { render, screen } from '@testing-library/react';
import MovieMedallionItem from './MovieMedallionItem';

describe('MOVIEMEDALLIONITEM SNAPSHOTS TESTS', () => {
    test('active MovieMedallionItem should not be changed', () => {
        render(
            <MovieMedallionItem text="Текст">
                <div></div>
            </MovieMedallionItem>,
        );

        const div = screen.getByTestId('medallionItem');
        expect(div).toMatchSnapshot();
    });

    test('inactive MovieMedallionItem should not be changed', () => {
        render(
            <MovieMedallionItem text="Текст" disabled={true}>
                <div></div>
            </MovieMedallionItem>,
        );
        const div = screen.getByTestId('medallionItem');
        expect(div).toMatchSnapshot();
    });
});
