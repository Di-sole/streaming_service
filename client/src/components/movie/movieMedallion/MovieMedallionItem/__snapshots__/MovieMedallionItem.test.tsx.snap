// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MOVIEMEDALLIONITEM SNAPSHOTS TESTS active MovieMedallionItem should not be changed 1`] = `
<div
  class="container"
  data-testid="medallionItem"
>
  <div
    class="container__content"
  >
    <div
      class="container__contentBlock"
    >
      <div />
    </div>
  </div>
  <div
    class="container__caption"
  >
    Текст
  </div>
</div>
`;

exports[`MOVIEMEDALLIONITEM SNAPSHOTS TESTS inactive MovieMedallionItem should not be changed 1`] = `
<div
  class="container container_inactive"
  data-testid="medallionItem"
>
  <div
    class="container__content"
  >
    <div
      class="container__contentBlock"
    >
      <div />
    </div>
  </div>
  <div
    class="container__caption"
  >
    Текст
  </div>
</div>
`;
